### PR TITLE
Fix weakremovers

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: pkglistgen/tool.py [hunk 1]
```diff
@@ -431,15 +431,17 @@ def create_weakremovers(self, target, target_config, directory, output):
 
                 pool.createwhatprovides()
 
+                accepted_archs = set(self.all_architectures)
+                accepted_archs.add('noarch')
+
                 for s in oldsysrepo.solvables_iter():
-                    if s.arch == 'src':
+                    if not s.arch in accepted_archs:
```
Comment (Vogtinator) [inline pkglistgen/tool.py:438]: `not in`

File: pkglistgen/tool.py [hunk 2]
```diff
@@ -431,15 +431,17 @@ def create_weakremovers(self, target, target_config, directory, output):
 
                 pool.createwhatprovides()
 
+                accepted_archs = set(self.all_architectures)
```
Comment (Vogtinator) [inline pkglistgen/tool.py:434]: Can this possibly have issues with `i686` vs `i586` (and similar flavors)?

Comment (coolo) [reply pkglistgen/tool.py:434]: we can swap the arch replacement with the check

File: pkglistgen/tool.py (hunk 1 above)
Comment (coolo) [reply pkglistgen/tool.py]: There are 127 of such uses - that's E713 btw

Comment (Vogtinator) [reply pkglistgen/tool.py]: Is it at least consistently wrong?

**Comment by Vogtinator** [comment]
Should work as designed, just can't tell for sure whether the design is correct...

**Pull request merged**